packages/admin/admin/src/dataGrid/muiGridFilterToGql.tsx

```diff
@@ -0,0 +1,74 @@
+import { GridColDef, GridFilterModel } from "@mui/x-data-grid";
+
+const muiGridOperatorValueToGqlOperator: { [key: string]: string } = {
+    contains: "contains",
+    equals: "equal",
+    ">": "greaterThan",
+    ">=": "greaterThanEqual",
+    "<": "lowerThan",
+    "<=": "lowerThanEqual",
+    "=": "equal",
+    "!=": "notEqual",
+    startsWith: "startsWith",
+    endsWith: "endsWith",
+    isAnyOf: "isAnyOf",
+    isEmpty: "isEmpty",
+    isNotEmpty: "isNotEmpty",
+    is: "equal",
+    not: "notEqual",
+    after: "greaterThan",
+    onOrAfter: "greaterThanEqual",
+    before: "lowerThan",
+    onOrBefore: "lowerThanEqual",
+};
+
+interface GqlStringFilter {
+    contains?: string | null;
+    startsWith?: string | null;
+    endsWith?: string | null;
+    equal?: string | null;
+    notEqual?: string | null;
+}
+interface GqlNumberFilter {
+    equal?: number | null;
+    lowerThan?: number | null;
+    greaterThan?: number | null;
+    lowerThanEqual?: number | null;
+    greaterThanEqual?: number | null;
+    notEqual?: number | null;
+}
+type GqlFilter = {
+    [key: string]: GqlStringFilter | GqlNumberFilter; //TODO add Boolean, Date, DateTime(?), SingleSelect(??)
+} & {
+    and?: GqlFilter[] | null;
+    or?: GqlFilter[] | null;
+};
+export function muiGridFilterToGql(columns: GridColDef[], filterModel?: GridFilterModel): { filter: GqlFilter; query?: string } {
+    if (!filterModel) return { filter: {} };
+    const filterItems = filterModel.items
+        .filter((value) => value.value !== undefined)
+        .map((value) => {
+            if (!value.operatorValue) throw new Error("operaturValue not set");
+            const gqlOperator = muiGridOperatorValueToGqlOperator[value.operatorValue];
+            if (!gqlOperator) throw new Error(`unknown operator ${value.operatorValue}`);
+            const column = columns.find((i) => i.field == value.columnField);
+            const type = column?.type;
+            const convertedValue = type === "number" ? parseFloat(value.value) : value.value;
+            return {
+                [value.columnField]: {
+                    [gqlOperator]: convertedValue,
+                } as GqlStringFilter | GqlNumberFilter,
+            };
+        });
+    const filter: GqlFilter = {};
+    const op: "and" | "or" = filterModel.linkOperator ?? "or";
+    filter[op] = filterItems;
+
+    let query: undefined | string = undefined;
+
+    if (filterModel.quickFilterValues) {
+        query = filterModel.quickFilterValues.join(" ");
```

Comment:
When i first saw this variable, in a story and the forwarding of `query` to the `variables` of a `useQuery` Hook i somehow was confused by the naming `query` because this naming somehow conflicts for me with apollo query. 

I probably would suggest renaming `filter` and `query` to something more explicit. i am still not 100% sure whats the difference between filter and query :-/

Reply:
hm, I see your point and I'm open for changes.

mui names it `filter` and `quickFilter`

Reply:
I already thought for a while about that. did not find a suitable naming, because i don't understand the need that we have to export filter and query. Why do we need both - is there a reason?

Reply:
> Why do we need both - is there a reason?

A primary difference is the scope of the filter: `query` is a simple string filter to perform an ilike query on _all_ fields, whereas `filter` is a collection of filters, where each filter is applied on a _specific_ field.

In my opinion `query` and `filter` is perfectly fine. There'll always be a conflict when using query: GraphQL queries, query params, SQL queries etc. We could name it `searchTerm` if we really need a differentiation.

We could also hide the return value of `muiGridFilterToGql` by directly spreading it into the query variables:

```typescript
const { data, loading, error } = useQuery(productsQuery, {
    variables: {
        ...muiGridFilterToGql(columns, dataGridProps.filterModel),
        offset: dataGridProps.page * dataGridProps.pageSize,
        limit: dataGridProps.pageSize,
        sort: muiGridSortToGql(sortModel),
    },
});
```

This removes the `query` and `filter` variable initialization – and may solve the confusion about naming

Reply:
ok, now I understand what this is doing :-)

yes - query is quite common for those kinds of things in APIs - agree. As I said for me, it was just confusing to use the `query` variable in a `useQuery` hook as a variable.


For me it is more like a `fullTextFilter` or `searchTerm`

Reply:
> This removes the query and filter variable initialization – and may solve the confusion about naming

yes, good idea.

what would be about `search` (instead of query)?

Reply:
yeah, `search` is quite nice - i like it

Reply:
> what would be about search (instead of query)?

👍🏼 

We'll need to change this in the api generator as well

Reply:
> We'll need to change this in the api generator as well

yes, luckily we don't have a 3.0 stable yet :)

Reply:
changed (also the api side in this PR)

Reply:
much better now!

Reply:
i also like it way much more. thanks for changing it :-)